lib/prius/registry.rb

```diff
@@ -73,5 +74,14 @@ def load_date(name, required)
     rescue ArgumentError
       raise TypeMismatchError, "'#{name}' value '#{value}' is not a date"
     end
+
+    def load_datetime(name, required)
+      value = load_string(name, required)
+      return nil if value.nil?
+
+      DateTime.parse(value)
```

Comment:
Driveby from me:

Looking at the [documentation](https://ruby-doc.org/stdlib-3.0.0/libdoc/date/rdoc/DateTime.html#method-c-parse):
> This method **does not** function as a validator. If the input string does not match valid formats strictly, you may get a cryptic result. Should consider to use `DateTime.strptime` instead of this method as possible.

Should we use a standardised format (eg. [`DateTime.rfc3339(...)`](https://ruby-doc.org/stdlib-3.0.0/libdoc/date/rdoc/DateTime.html#method-c-rfc3339)) to avoid ambiguity?

Reply:
Optionally, we can call the arguument `:datetime_rfc3339` if we want.

Reply:
Petition to use `ActiveSupport::Datetime`. It has proper timezone support, and we can safely (de)serialize values with the `iso8601` method which is our general standard elsewhere (e.g. in Pub/Sub messages).

Reply:
I came here to talk about the above Date.parse or DateTime.parse are Russian roulettes so we def need to use a specific format and agree on it.

Reply:
> ActiveSupport::TimeWithZone

that would mean adding rails as a dependency, not sure we should do that.

Reply:
@kevinrobayna I think we can just add `active_support` as a dependency and require the minimal `active_support` requirements and `active_support/time_with_zone` like so:

```ruby
require "active_support"
require "active_support/core_ext/hash/indifferent_access"
```

Reply:
that is "rails" 😅

Reply:
So I just found this:
> [DateTime](https://rubyapi.org/3.3/o/datetime) class is considered deprecated. Use [Time](https://rubyapi.org/3.3/o/time) class.

I'm in two minds about adding a big dependency to a very minimal gem as well. Granted that `TimeWithZone` comes with a bunch of goodies, but looks like `Time` does handle the basics of timezones correctly:
```ruby
irb(main):001> require "time"
=> true
irb(main):002> t = Time.iso8601("2023-10-07T21:15:30+11:00")
=> 2023-10-07 21:15:30 +1100
irb(main):003> t.utc
=> 2023-10-07 10:15:30 UTC
```
Plus if this gem is used in the context of a Rails app, then we can just do `t.in_time_zone` to get a full-fledged `ActiveSupport::TimeWithZone`, so probably not too bad?